solr/core/src/java/org/apache/solr/core/SolrCore.java

```diff
@@ -1899,6 +1899,18 @@ public boolean isClosed() {
     return refCount.get() <= 0;
   }
 
+  /**
+   * Whether this core is ready. Sometimes using {@link #isClosed()} might result in a deadlock due
+   * to the caller trying to eagerly open a new searcher via {@link #withSearcher(IOFunction)}
+   * during core init phase.
+   *
+   * <p>Like for example concurrently polling metrics collection via the registered gauges in {@link
+   * #initializeMetrics(SolrMetricsContext, String)}
```

Comment:
Personally, I think these docs are overly prescriptive based on our first need for this.  Really, the docs can be as simple as: "The core is ready for use.  It is not initializing or closing/closed.